net/netns/netns_linux.go

```diff
@@ -135,6 +136,11 @@ func setBypassMark(fd uintptr) error {
 func bindToDevice(fd uintptr) error {
 	ifc, err := defaultRouteInterface()
 	if err != nil {
+		if runtime.GOOS == "android" {
+			// Some Android devices do not tolerate binding to "lo",
```

Comment:
What's the actual err value you get? And from defaultRouteInterface or did you get an error later at the actual binding?

I think recent Androids blocked access to `/proc/net/route`. (I've worked around that elsewhere, in `net/interfaces`'s `likelyHomeRouterIPAndroid`)

I worry about this causing routing loop issues if we don't get this route, especially as we add default route support in the future.

Reply:
I get "no default routes found" from defaultRouteInterface. No error from SetsockoptString (when this change is not applied).

Reply:
What error was SO_BINDTODEVICE giving you before this change, when it was passing the dummy string `"lo"` as the `SetsockoptString` value?

Reply:
cc @danderson

Reply:
SO_BINDTODEVICE gave no error, probably because the interface exists:

```
$ adb shell ip route show table 0
default via 192.168.86.1 dev wlan0  table 1025  proto static 
192.168.86.0/24 dev wlan0  table 1025  proto static  scope link 
default dev dummy0  table 1002  proto static  scope link 
192.168.86.0/24 dev wlan0  proto kernel  scope link  src 192.168.86.27 
broadcast 127.0.0.0 dev lo  table local  proto kernel  scope link  src 127.0.0.1 
local 127.0.0.0/8 dev lo  table local  proto kernel  scope host  src 127.0.0.1 
local 127.0.0.1 dev lo  table local  proto kernel  scope host  src 127.0.0.1 
broadcast 127.255.255.255 dev lo  table local  proto kernel  scope link  src 127.0.0.1 
broadcast 192.168.86.0 dev wlan0  table local  proto kernel  scope link  src 192.168.86.27 
local 192.168.86.27 dev wlan0  table local  proto kernel  scope host  src 192.168.86.27 
broadcast 192.168.86.255 dev wlan0  table local  proto kernel  scope link  src 192.168.86.27 
fe80::/64 dev wlan0  table 1025  proto kernel  metric 256  pref medium
fe80::/64 dev wlan0  table 1025  proto static  metric 1024  pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
fe80::/64 dev p2p0  table 1026  proto kernel  metric 256  pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
fe80::/64 dev dummy0  table 1002  proto kernel  metric 256  pref medium
default dev dummy0  table 1002  proto static  metric 1024  pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
local ::1 dev lo  table local  proto none  metric 0  pref medium
local fe80::c47:86ff:fe11:ebe4 dev lo  table local  proto none  metric 0  pref medium
local fe80::6cb7:49ff:fecc:4270 dev lo  table local  proto none  metric 0  pref medium
local fe80::6eb7:49ff:fecc:4270 dev lo  table local  proto none  metric 0  pref medium
ff00::/8 dev dummy0  table local  metric 256  pref medium
ff00::/8 dev p2p0  table local  metric 256  pref medium
ff00::/8 dev wlan0  table local  metric 256  pref medium
unreachable default dev lo  proto kernel  metric 4294967295  error -101 pref medium
```

I'm working on adding Android support to defaultRouteInterface. Question is, should "lo" still be the fallback on Andoid? I suppose non-functional dials is better than routing loops.

Reply:
FWIW, `/proc/net/route` is available, but doesn't list a route with mask 0:

```
$ adb shell cat /proc/net/route
Iface   Destination     Gateway         Flags   RefCnt  Use     Metric  Mask            MTU     Window  IRTT                                                       
wlan0   0056A8C0        00000000        0001    0       0       0       00FFFFFF        0       0       0            
```